pkg/spec/spec.go

```diff
@@ -21,7 +21,10 @@ const cpuPeriod = 100000
 // CreateConfigToOCISpec parses information needed to create a container into an OCI runtime spec
 func CreateConfigToOCISpec(config *CreateConfig) (*spec.Spec, error) { //nolint
 	cgroupPerm := "ro"
-	g := generate.New()
+	g, err := generate.New("linux")
```

Comment:
The rest of `CreateConfigToOCISpec` is pretty Linux-specific, so I'm fine with this being hard-coded here.  But `CreateConfigToOCISpec` is so opinionated, I'm surprised you need to seed it at all.  Do you know what parts of this initial config survive the subsequent edits to be returned to the caller?  You might just want to build the `*spec.Spec` object directly.